### PR TITLE
feat: Skeleton loading #404

### LINKED ISSUE #404: Skeleton loading
Add shimmers to components for a nicer loading experience
```[tasklist]
### Tasks
- [ ] https://github.com/twentyhq/twenty/issues/434
```

### PR DESCRIPTION
**Pull request by deepaksing**
this PR fixes #404 

- Used  **react-loading-skeleton** library to add skeleton loading.

**Result:**
image

### CODE REVIEW

File: front/src/modules/relation-picker/components/SingleEntitySelect.tsx [hunk 1]
```diff
@@ -16,6 +17,8 @@ import { isDefined } from '@/utils/type-guards/isDefined';
 
 import { useEntitySelectLogic } from '../hooks/useEntitySelectLogic';
 
+import 'react-loading-skeleton/dist/skeleton.css';
```
Comment (charlesBochet) [inline front/src/modules/relation-picker/components/SingleEntitySelect.tsx:20]: I would import it in src/index.tsx as skeletons is a global feature and not tied to the SingleEntitySelect

File: front/src/modules/relation-picker/components/SingleEntitySelect.tsx [hunk 2]
```diff
@@ -97,7 +100,9 @@ export function SingleEntitySelect<
           </DropdownMenuSelectableItem>
         ))}
         {entitiesInDropdown?.length === 0 && (
-          <DropdownMenuItem>No result</DropdownMenuItem>
+          <DropdownMenuItem>
```
Comment (charlesBochet) [inline front/src/modules/relation-picker/components/SingleEntitySelect.tsx:103]: this should not replace the "No result" section.
There are actual cases when data is loaded but there is no result

**Comment by charlesBochet** [comment]
@deepaksing Thanks a lot for your contributions!

We would like the skeleton to look like the actual component. So here I think we need to introduce a `DropdownMenuItemContainerSkeleton` that would mimic the `DropdownMenuItemContainer` UI. See this blog post: https://birdeatsbug.com/blog/implementing-skeleton-screen-in-react-with-react-loading-skeleton-and-suspense with the `CardSkeleton` example.

I would also look into the react `Suspense` component to see if we could leverage it!

**Comment by deepaksing** [comment]
@charlesBochet, Sure i will try this

**Comment by charlesBochet** [comment]
@deepaksing As discussed with you on Slack, let's not use Suspense and let's use the return { loading } of the query to display the DropdownMenuItemContainerSkeleton directly in the component

File: front/src/index.tsx [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import ReactDOM from 'react-dom/client';
+import Skeleton from 'react-loading-skeleton';
```
Comment (charlesBochet) [inline front/src/index.tsx:3]: I don't think this import is needed!

File: front/src/modules/relation-picker/components/SingleEntitySelect.tsx [hunk 4]
```diff
@@ -80,24 +83,29 @@ export function SingleEntitySelect<
...
-              avatarUrl={entity.avatarUrl}
-              placeholder={entity.name}
-              size={16}
-              type={entity.avatarType ?? 'rounded'}
-            />
-            {entity.name}
-          </DropdownMenuSelectableItem>
-        ))}
-        {entitiesInDropdown?.length === 0 && (
+        {entities.loading == true ? (
```
Comment (charlesBochet) [inline front/src/modules/relation-picker/components/SingleEntitySelect.tsx:86]: { entities.loading ? (

File: front/src/modules/ui/components/menu/DropdownMenuItemContainerSkeleton.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import styled from '@emotion/styled';
```
Comment (charlesBochet) [inline front/src/modules/ui/components/menu/DropdownMenuItemContainerSkeleton.tsx:1]: Please move this file to relation-picker/components/skeletons

File: front/src/modules/ui/components/skeletons/CompanyPickerSkeleton.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import Skeleton from 'react-loading-skeleton';
```
Comment (charlesBochet) [inline front/src/modules/ui/components/skeletons/CompanyPickerSkeleton.tsx:1]: Please move this file to relation-picker/components/skeletons

**Review by charlesBochet — APPROVED** [review]
Thanks for adding a first skeleton loader! This will set the path for other ones!

**Pull request merged**